simplemma/lemmatizer.py

```diff
@@ -134,20 +134,22 @@ def _dehyphen(token: str, datadict: Dict[str, str]) -> Optional[str]:
     splitted = HYPHEN_REGEX.split(token)
     if len(splitted) <= 1 or not splitted[-1]:
         return None
+
     # try to find a word form without hyphen
-    subcandidate = "".join([t for t in splitted if t not in HYPHENS]).lower()
+    subcandidate = "".join(t for t in splitted if t not in HYPHENS).lower()
     if token[0].isupper():
         subcandidate = subcandidate.capitalize()
-    if subcandidate in datadict:
-        return datadict[subcandidate]
+    subcandidate_lemma = _simple_search(subcandidate, datadict)
```

Comment:
This is more costly and not useful IMO, it looks for different combinations although only "subcandidate" in its current write form is useful.